plugins/inventory/now.py

```diff
@@ -0,0 +1,443 @@
+#  |  |--FileServerFloor1
+#  |  |--FileServerFloor2
+#  |--@Linux_Red_Hat:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |--@Windows_XP:
+#  |  |--FileServerFloor1
+#  |  |--FileServerFloor2
+#  |  |--INSIGHT-NY-03
+#  |--@ungrouped:
+
+
+# Group hosts into named groups, according to the specified criteria.
+# Below example creates a single group containing hosts that match
+# all the criteria.
+groups:
+  non_windows_prod_servers:
+    classification:
+      includes: [ Production ]
+    os:
+      excludes:
+        - Windows XP
+        - Windows 2000
+        - Windows 2000 Server
+        - Windows 2003 Standard
+
+# `ansible-inventory -i inventory.now.yaml --graph` output:
+# @all:
+#  |--@HP_UX:
+#  |  |--Car-2
+#  |--@Linux_Red_Hat:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |  |--lnux100
+#  |  |--lnux101
+#  |--@Solaris:
+#  |  |--MailServerUS
+#  |--@ungrouped:
+
+
+# Configure inventory host names and host vars.
+plugin: servicenow.itsm.now
+columns:
+  - name
+  - classification
+  - cpu_type
+ansible_host_source: fqdn
+inventory_hostname_source: asset_tag
+
+# `ansible-inventory -i inventory.now.yaml --graph --vars` output:
+# @all:
+#  |--@ungrouped:
+#  |  |--P1000019
+#  |  |  |--{ansible_host = my.server.com}
+#  |  |  |--{classification = Production}
+#  |  |  |--{cpu_type = Intel}
+#  |  |  |--{name = SAP-SD-02}
+"""
+
+
+class InventoryModule(BaseInventoryPlugin):
+
+    NAME = "servicenow.itsm.now"
+
+    def verify_file(self, path):
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(("now.yaml", "now.yml")):
+                return True
+            # TODO remove?
+            self.display.vvv(
+                'Skipping due to inventory source not ending in "now.yaml" nor "now.yml"'
+            )
+            return False
+
+    def validate_grouping_conditions(self, parent):
+        for category, val in parent.items():
+            if val and all(i in val for i in ["includes", "excludes"]):
+                raise AnsibleParserError(
+                    "Invalid configuration: 'includes' and 'excludes' are mutually exclusive."
+                )
+
+    def _includes_query(self, column, includes):
+        return "^OR".join("{0}={1}".format(column, i) for i in includes)
+
+    def _excludes_query(self, column, excludes):
+        return "^".join("{0}!={1}".format(column, i) for i in excludes)
+
+    def _sysparm_query(self, conditions):
+        param_queries = []
+        for column, val in conditions.items():
+            if val:
+                includes = val.get("includes")
+                if includes:
+                    param_queries.append(self._includes_query(column, includes))
+                excludes = val.get("excludes")
+                if excludes:
+                    param_queries.append(self._excludes_query(column, excludes))
+        if param_queries:
+            return "^".join(i for i in param_queries)
+        return None
```

Comment:
Maybe we should make this method public and provide some unit tests that will serve as a source of examples?